src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameNodeDefinitions.cs

```diff
@@ -0,0 +1,375 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.Common;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.StackFrame
+{
+    using StackFrameNodeOrToken = EmbeddedSyntaxNodeOrToken<StackFrameKind, StackFrameNode>;
+    using StackFrameToken = EmbeddedSyntaxToken<StackFrameKind>;
+    using StackFrameTrivia = EmbeddedSyntaxTrivia<StackFrameKind>;
+
+    internal abstract class StackFrameNode : EmbeddedSyntaxNode<StackFrameKind, StackFrameNode>
+    {
+        protected StackFrameNode(StackFrameKind kind) : base(kind)
+        {
+        }
+
+        public abstract void Accept(IStackFrameNodeVisitor visitor);
+    }
+
+    /// <summary>
+    /// Root of all expression nodes.
+    /// </summary>
+    internal abstract class StackFrameExpressionNode : StackFrameNode
+    {
+        protected StackFrameExpressionNode(StackFrameKind kind) : base(kind)
+        {
+        }
+    }
+
+    internal sealed class StackFrameMethodDeclarationNode : StackFrameExpressionNode
```

Comment:
not a fan of 'Declaration' deriving from Expression.

Reply:
Changed to just derive from `StackFrameNode`